src/AI.java

```diff
@@ -4,8 +4,8 @@
 public class AI extends Player{
     private static int id = 1;
     private static final int DRAW = -1;
-    public AI (){
-        super("Bot " + id);
+    public AI (Deck deck){
+        super("Bot " + id, deck);
```

Comment:
Why do you need to pass deck into the constructor? Doesn't this class inherit the methods that interact with deck from palyer?